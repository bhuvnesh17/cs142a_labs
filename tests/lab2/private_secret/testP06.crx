// test parameters

func main() : void {
  let a = 7;
}

func main(a : void) : void {
  let a = 7;
}

func main(a: void, b:int) : void {
  let a =  7;
}

func main(a:void, b:int, c:float) : void {
  let a = 7;
}

func main(a:void, b:int, c:float, d:bool): void {
  let a = 7;
}

func main(main : grue, b : gorn, c:int, d:void) : void {
  let a = 7;
}
