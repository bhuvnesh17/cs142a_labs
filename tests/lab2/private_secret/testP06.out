PROGRAM
  DECLARATION_LIST
    DECLARATION
      FUNCTION_DEFINITION
        PARAMETER_LIST
        TYPE
        STATEMENT_BLOCK
          STATEMENT_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
    DECLARATION
      FUNCTION_DEFINITION
        PARAMETER_LIST
          PARAMETER
            TYPE
        TYPE
        STATEMENT_BLOCK
          STATEMENT_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
    DECLARATION
      FUNCTION_DEFINITION
        PARAMETER_LIST
          PARAMETER
            TYPE
          PARAMETER
            TYPE
        TYPE
        STATEMENT_BLOCK
          STATEMENT_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
    DECLARATION
      FUNCTION_DEFINITION
        PARAMETER_LIST
          PARAMETER
            TYPE
          PARAMETER
            TYPE
          PARAMETER
            TYPE
        TYPE
        STATEMENT_BLOCK
          STATEMENT_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
    DECLARATION
      FUNCTION_DEFINITION
        PARAMETER_LIST
          PARAMETER
            TYPE
          PARAMETER
            TYPE
          PARAMETER
            TYPE
          PARAMETER
            TYPE
        TYPE
        STATEMENT_BLOCK
          STATEMENT_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL
    DECLARATION
      FUNCTION_DEFINITION
        PARAMETER_LIST
          PARAMETER
            TYPE
          PARAMETER
            TYPE
          PARAMETER
            TYPE
          PARAMETER
            TYPE
        TYPE
        STATEMENT_BLOCK
          STATEMENT_LIST
            STATEMENT
              ASSIGNMENT_STATEMENT
                DESIGNATOR
                EXPRESSION0
                  EXPRESSION1
                    EXPRESSION2
                      EXPRESSION3
                        LITERAL

